pandas/tests/io/excel/test_readers.py

```diff
@@ -1499,3 +1499,12 @@ def test_engine_invalid_option(self, read_ext):
         with pytest.raises(ValueError, match="Value must be one of *"):
             with pd.option_context(f"io.excel{read_ext}.reader", "abc"):
                 pass
+
+    @td.check_file_leaks
+    def test_corrupt_files_closed(self, request, engine, read_ext):
+        # GH41778
+        msg = "File is not a zip file"
+        with tm.ensure_clean(f"corrupt{read_ext}") as file:
+            with pytest.raises((BadZipFile, ValueError), match=msg):
+                with pd.ExcelFile(file, engine=engine) as _:
+                    pass
```

Comment:
this is testing the message from the `pd.ExcelFile(file, engine=engine) as _` expression which does already raise `BadZipFile: File is not a zip file`

how is this testing that the file is closed?

Reply:
I can't replicate the issue locally to see this test fail without the patch.

I assume it's Windows specific (I use windows, but use wsl and don't have dev tools installed in native windows)

I think that ensure_clean is supposed to remove files, closed or not, so the file leaks check is checking ensure_clean and not the op under test?

Reply:
The test worked only on windows: `tm.ensure_clean` will fail to remove the file and throw an error (the file leak check doesn't catch the `ResourceWarning`).

Locally, the test fails for me now when the test is not applied (but it might depend on whether `ResourceWarnings` are en/disabled).